python/pyspark/ml/pipeline.py

```diff
@@ -206,49 +167,65 @@ def save(self, path):
     @classmethod
     @since("2.0.0")
     def read(cls):
-        """Returns an JavaMLReader instance for this class."""
+        """Returns an MLReader instance for this class."""
         return PipelineMLReader(cls)
 
     @classmethod
-    @since("2.0.0")
-    def load(cls, path):
-        """Reads an ML instance from the input path, a shortcut of `read().load(path)`."""
-        return cls.read().load(path)
+    def _transfer_stage_from_java(cls, java_stage):
```

Comment:
It is a little confusing to call it `java_stage` while the expected input is a pipeline. Shall we rename it to `_from_java_pipeline` and `_transfer_stage_to_java` to `_to_java_pipeline`? If we have to use the same method name, I would recommend `_from_java` and `_to_java` instead.